internal/host/statemachine.go

```diff
@@ -593,8 +593,9 @@ func NewHostStateMachine(sm stateswitch.StateMachine, th *transitionHandler) sta
 		},
 		Condition: stateswitch.And(If(IsConnected), If(IsMediaConnected), If(HasInventory),
 			hasMinRequiredHardware,
-			requiredInputFieldsExist,
-			stateswitch.Not(isSufficientForInstall)),
+			stateswitch.Or(stateswitch.Not(If(VSphereHostUUIDEnabled)),
```

Comment:
To see if i understand correctly, this is the only transition that actually change? 
Why the or include `requiredInputFieldsExist`? shouldn't it be only `stateswitch.Not(If(VSphereHostUUIDEnabled)` and ` stateswitch.Not(isSufficientForInstall)`?

Reply:
if requiredInputFieldsExist has failed (missing machine CIDR) the status should be pending for input unless VSphereHostUUIDEnabled check failed (the status should be insufficient)

Reply:
The second `requiredInputFieldsExist`  (In the last `And` condition), should be removed because it is always true

Reply:
Why not add `VSphereHostUUIDEnabled` to `isSufficientForInstall` and leave the rules as before?

Reply:
How about adding `HostStatusPendingForInput` to  https://github.com/openshift/assisted-service/blob/f40267245a6bc7e170b4e6acb368b7c41d8f0397/internal/host/statemachine.go#L550-L562 and returning `VSphereHostUUIDEnabled` to be part of `hasMinRequiredHardware` ?

Reply:
or moving it to `isSufficientForInstall` as Ori suggested

Reply:
if Vsphere validation will be part of isSufficientForInstall and the cluster doesn't have machine cidr the status will be "Pending for input" instead of "insufficient".
Adding it to hasMinRequiredHardware (as i did before this PR) prevent us from integration with vsphere (and updating the host in general) since there is no transition for  "Pending for input" and hasMinRequiredHardware validation failed.
The ultimate solution is to add the vsphere condition to hasMinRequiredHardware  and add "pending for input" to the status list of the following transition.

```
sm.AddTransition(stateswitch.TransitionRule{
		TransitionType: TransitionTypeRefresh,
		SourceStates: []stateswitch.State{
			stateswitch.State(models.HostStatusDisconnected),
			stateswitch.State(models.HostStatusDiscovering),
			stateswitch.State(models.HostStatusInsufficient),
			stateswitch.State(models.HostStatusKnown),
		},
		Condition: stateswitch.And(If(IsConnected), If(IsMediaConnected), If(HasInventory),
			stateswitch.Not(hasMinRequiredHardware)),
		DestinationState: stateswitch.State(models.HostStatusInsufficient),
		PostTransition:   th.PostRefreshHost(statusInfoInsufficientHardware),
	})
```

Reply:
Implementing the ultimate solution is risky for production.
As a short term solution i covered the vsphere case only.
Afterward we should implement the ultimate solution that will resolve any modification in Pending for input status.

Reply:
Did you try to test https://github.com/openshift/assisted-service/pull/4061/files#r913489952? 
Is any of the existing tests breaks? does a regression pass?